content/post/debrick-core9g25-intro.md

```diff
@@ -0,0 +1,131 @@
+quite old we reflash the NAND and replace the stack with more recent versions
+of the programs. Reflashing the NAND is dangerous and can lead to bricked and
+unusable devices. In a series of 3 posts I'll demonstrate how to fix a bricked
+CORE 9G25. In this first post I'll explain how normal reflash procudere and
+boot procedure of the CORE 9G25 work. The second post will be about reflashing
+the NAND using the tool SAM-BA on Linux. The third post will be about the same
+thing, but then Windows is used to fix the NAND.
+
+## Reflash procedure
+Reflashing the content on the Nand is easy when all goes well. During boot, the
+bootloader U-boot look for an USB drive to be present. When it's found so
+U-Boot will look for files with specific names, load them into RAM before writing
+them to specific segments of the NAND flash. So if you want to replace the Linux
+kernel you've to make sure to put a file called `uImage` on the USB drive.
+The snippet below shows the output of the U-Boot script that replaces the
+content on the NAND.
+
+```
+1 Storage Device(s) found
+
+########## Reading at91sam9x5ek-nandflashboot-uboot-3.5.3.bin ###########
+
+#### Read 0x2FB0(12208) bytes ###
+Erase NAND flash 0x0 0x40000 OK
+
+########## Reading u-boot.bin ###########
+##
+#### Read 0x6BD04(441604) bytes ###
+Erase NAND flash 0x40000 0x80000 OK
+
+########## Reading at91sam9g25ek.dtb ###########
+
+#### Read 0x5C42(23618) bytes ###
+Erase NAND flash 0x180000 0x80000 OK
+
+########## Reading uImage ###########
+####################
+#### Read 0x304FDA(3166170) bytes ###
+Erase NAND flash 0x200000 0x600000 OK
+
+########## Reading rootfs.img ###########
+####################
+#### Read 0xE40000(14942208) bytes ###
+Erase NAND flash 0x800000 0xf800000 OK
+
+### Update OK ### Please RESET the board ###
+```
+
+As you can see U-Boot updates the NAND using 5 files.
+
+* `at91sam9x5ek-nandflashboot-uboot-3.5.3.bin` - the bootloader
+[AT91Bootstrap][at91bootstrap]
+* `u-boot.bin` - the bootloader U-Boot
+* `at91sam9g25ek.dtb` - the device tree blob
+* `uImage` - the Linux kernel
+* `rootfs.img` - the root filesystem
+
+First U-Boot loads the file `at91sam9x5ek-nandflashboot-uboot-3.5.3.bin`, than
+it ereases a block of `0x40000` bytes on the NAND, starting at `0x0`. It then
+writes the file to Nand, aligning the program at address `0x0`.
+
+## Risks
+Because U-Boot performs the update, updating U-Boot is risky.  What happens if
+the binaries for AT91Boostrap or U-Boot aren't compiled correctly? Or what if
+the power is lost during the update?  Or what if you've replaced the original
+U-Boot with a version that doesn't contain the update script (yes, I've done
+that)? Well, you're devices bricked.
+
+Only recently together my college Marten van Houten and I managed to reflash
+the NAND using tools provided but the vendor of the CPU.
+
+In order to understand fully how we managed to do that I'll explain the boot
```

Comment:
understand fully -> fully understand
`how we managed to do that` -> how we managed it` first one is slightly ambiguous